src/PermissionRegistrar.php

```diff
@@ -17,39 +17,27 @@ class PermissionRegistrar
     /** @var \Illuminate\Contracts\Cache\Repository */
     protected $cache;
 
-    /** @var \Illuminate\Contracts\Logging\Log */
-    protected $logger;
-
     /** @var string */
     protected $cacheKey = 'spatie.permission.cache';
 
-    public function __construct(Gate $gate, Repository $cache, Log $logger)
+    public function __construct(Gate $gate, Repository $cache)
     {
         $this->gate = $gate;
         $this->cache = $cache;
-        $this->logger = $logger;
     }
 
     public function registerPermissions(): bool
     {
-        try {
-            $this->getPermissions()->map(function (Permission $permission) {
-                $this->gate->define($permission->name, function ($user) use ($permission) {
-                    return $user->hasPermissionTo($permission);
-                });
-            });
-
-            return true;
-        } catch (Exception $exception) {
-            if ($this->shouldLogException()) {
-                $this->logger->alert(
-                    "Could not register permissions because {$exception->getMessage()}".PHP_EOL.
-                    $exception->getTraceAsString()
-                );
+        $this->gate->before(function (Authenticatable $user, string $ability) {
+            try {
+                if (method_exists($user, 'hasPermissionTo')) {
+                    return $user->hasPermissionTo($ability);
+                }
+            } catch (PermissionDoesNotExist $e) {
             }
+        });
 
-            return false;
-        }
+        return true;
```

Comment:
Is this `return true;` needed?

I know it used to return a boolean that was checked in dozens of tests (never understood why that was done), but it appears you've removed that check from the tests, so this `true` may not be needed.

Reply:
Update: I see, you've declared its return type as boolean, so it's gotta return "something" boolean.

Reply:
The `registerPermissions` method used to return `true` on success and `false` if an exception was thrown. Useful or not, chances are people are using this in their code and thus changing it would be breaking (which is what I wanted avoid).

Reply:
I'll accept that. 
Maybe consider a change in v3 if suitable then.

Reply:
Absolutely :)